traits/trait_types.py

```diff
@@ -1546,10 +1559,6 @@ class Directory(BaseDirectory):
     def __init__(
         self, value="", auto_set=False, entries=0, exists=False, **metadata
     ):
-        # Define the C-level fast validator to use if the directory existence
-        #: test is not required:
-        if not exists:
-            self.fast_validate = (ValidateTrait.coerce, str)
         super(Directory, self).__init__(
```

Comment:
We should probably have a comment indicating that fast validation is disabled. Ideally, also open an issue on the tracker to remind us to decide whether to implement fast validation or not.